src/Compilers/VisualBasic/Portable/Binding/Binder_Imports.vb

```diff
@@ -90,56 +90,53 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
+                ' Parser checks for type characters on alias text, so don't need to check again here.
+
+                ' Check for duplicate symbol.
+                If data.Aliases.ContainsKey(aliasText) Then
+                    Binder.ReportDiagnostic(diagBag, aliasIdentifier, ERRID.ERR_DuplicateNamedImportAlias1, aliasText)
+                Else
+                    ' Make sure that the Import's alias doesn't have the same name as a type or a namespace in the global namespace
+                    Dim conflictsWith = binder.Compilation.GlobalNamespace.GetMembers(aliasText)
+
+                    If Not conflictsWith.IsEmpty Then
+                        ' TODO: Note that symbol's name in this error message is supposed to include Class/Namespace word at the beginning.
+                        '       Might need to use special format for that parameter in the error message. 
+                        Binder.ReportDiagnostic(diagBag,
                                                     aliasImportSyntax,
                                                     ERRID.ERR_ImportAliasConflictsWithType2,
                                                     aliasText,
                                                     conflictsWith(0))
-                        Else
-                            ' NOTE: we are mimicing Dev10 behavior where referencing type symbol in 
-                            '       alias 'declaration' fails in case the symbol has errors.
+                    Else
+                        If aliasTarget.Kind <> SymbolKind.ErrorType Then
                             Dim useSiteErrorInfo As DiagnosticInfo = aliasTarget.GetUseSiteErrorInfo()
-                            If Not AllowAliasWithUseSiteError(useSiteErrorInfo) Then
-                                Binder.ReportDiagnostic(diagBag, aliasImportSyntax, useSiteErrorInfo)
-                                ' NOTE: Do not create an alias!!!
-
-                            Else
-                                Dim aliasSymbol = New AliasSymbol(binder.Compilation,
-                                                                 binder.ContainingNamespaceOrType,
-                                                                 aliasText,
-                                                                 aliasTarget,
-                                                                 If(binder.BindingLocation = BindingLocation.ProjectImportsDeclaration, NoLocation.Singleton, aliasIdentifier.GetLocation()))
 
-                                data.AddAlias(binder.GetSyntaxReference(aliasImportSyntax), aliasText, aliasSymbol, aliasImportSyntax.SpanStart)
+                            If ReportUseSiteErrorForAlias(useSiteErrorInfo) Then
+                                Binder.ReportDiagnostic(diagBag, aliasImportSyntax, useSiteErrorInfo)
                             End If
                         End If
+
+                        Dim aliasSymbol = New AliasSymbol(binder.Compilation,
```

Comment:
I am surprised to see us creating a new alias symbol here. Isn't there one already in existence?